validate/validate.go

```diff
@@ -139,78 +148,83 @@ func (v *Validator) CheckRoot() (msgs []string) {
 		rootfsPath = filepath.Join(v.bundlePath, v.spec.Root.Path)
 		absRootPath, err = filepath.Abs(rootfsPath)
 		if err != nil {
-			msgs = append(msgs, fmt.Sprintf("unable to convert %q to an absolute path", rootfsPath))
+			errs = multierror.Append(errs, fmt.Errorf("unable to convert %q to an absolute path", rootfsPath))
 		}
 	}
 
 	if fi, err := os.Stat(rootfsPath); err != nil {
-		msgs = append(msgs, fmt.Sprintf("Cannot find the root path %q", rootfsPath))
+		errs = multierror.Append(errs,
+			NewError(PathExistence, fmt.Sprintf("Cannot find the root path %q", rootfsPath), rspec.Version))
 	} else if !fi.IsDir() {
-		msgs = append(msgs, fmt.Sprintf("The root path %q is not a directory.", rootfsPath))
+		errs = multierror.Append(errs,
+			NewError(PathExistence, fmt.Sprintf("The root path %q is not a directory.", rootfsPath), rspec.Version))
 	}
 
 	rootParent := filepath.Dir(absRootPath)
```

Comment:
We will need a guard for the case where `absRootPath` was not initialized because the `Abs` call failed above.  Although that issue predates this PR and can be handled in a separate PR if you like.

Reply:
From the golang document, it seems only a terrible system failure will cause an Abs Error.
I think we can just `return`  when we catch an Abs error. I'll add it in the following update.